api/src/routes/settings.ts

```diff
@@ -163,6 +166,76 @@ export const settingRoutes: FastifyPluginCallbackTypebox = (
     }
   );
 
+  fastify.put(
+    '/update-my-username',
+    {
+      schema: {
+        body: Type.Object({
+          username: Type.String({ minLength: 3, maxLength: 20 })
+        }),
+        response: {
+          200: Type.Object({
+            message: Type.String(),
+            type: Type.Union([Type.Literal('success'), Type.Literal('info')])
+          }),
+          500: Type.Object({
+            message: Type.String(),
+            type: Type.Literal('danger')
+          })
+        }
+      }
+    },
+    async (req, reply) => {
+      try {
+        const user = await fastify.prisma.user.findFirst({
+          where: { id: req.session.user.id }
+        });
+
+        const newUsername = req.body.username.toLocaleLowerCase();
+        const oldUsername = user?.username.toLocaleLowerCase();
+
+        if (newUsername == oldUsername) {
+          return { message: 'flash.username-used', type: 'info' } as const;
+        }
```

Comment:
`usernameDisplay` is not compared with current one

[Old API Code](https://github.com/freeCodeCamp/freeCodeCamp/blob/129f629e10973c31109449dcfad668d5b0bd2190/api-server/src/server/boot/settings.js#L179-L181)